lib/Api/cloud_services.dart

```diff
@@ -10,10 +11,15 @@ class ApiConfig {
 }
 
 class CloudServices {
+
   final storage = SecureStorage();
   Future<bool> findUser(Map<String, dynamic> data) async {
     bool found = false;
-    final dio = Dio(BaseOptions(baseUrl: ApiConfig.baseUrl));
+    final dio = Dio(BaseOptions(
+      baseUrl: ApiConfig.baseUrl,
+      validateStatus: (status) => status! < 500,
+    ));
```

Comment:
nit: Adding this line to make 4xx as valid response.
Dio treats 4xx as an exception, so we cannot show exact error to the user. In our case 404 is used when the user is not found in the records which must be communicated to the user.